app/screens/settings/display_clock/display_clock.tsx

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect, useMemo, useState} from 'react';
+import {useIntl} from 'react-intl';
+import {StatusBar, View} from 'react-native';
+import {Edge, SafeAreaView} from 'react-native-safe-area-context';
+
+import {savePreference} from '@actions/remote/preference';
+import Block from '@components/block';
+import OptionItem from '@components/option_item';
+import {Preferences} from '@constants';
+import {useServerUrl} from '@context/server';
+import {useTheme} from '@context/theme';
+import useAndroidHardwareBackHandler from '@hooks/android_back_handler';
+import useNavButtonPressed from '@hooks/navigation_button_pressed';
+import {t} from '@i18n';
+import {popTopScreen, setButtons} from '@screens/navigation';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+
+const footer = {
+    id: t('settings_display.clock.preferTime'),
+    defaultMessage: 'Select how you prefer time displayed.',
+};
+
+const edges: Edge[] = ['left', 'right'];
+const CLOCK_TYPE = {
+    NORMAL: 'NORMAL',
+    MILITARY: 'MILITARY',
+} as const;
+
+const getStyleSheet = makeStyleSheetFromTheme((theme) => {
+    return {
+        container: {
+            flex: 1,
+            backgroundColor: theme.centerChannelBg,
+        },
+        wrapper: {
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.06),
+            flex: 1,
+            paddingTop: 35,
+        },
+        divider: {
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.1),
+            height: 1,
+        },
+        containerStyle: {
+            paddingHorizontal: 8,
+        },
+    };
+});
+
+const SAVE_CLOCK_BUTTON_ID = 'settings_display.clock.save.button';
+
+type DisplayClockProps = {
+    componentId: string;
+    currentUserId: string;
+    hasMilitaryTimeFormat: boolean;
+}
+const DisplayClock = ({componentId, currentUserId, hasMilitaryTimeFormat}: DisplayClockProps) => {
+    const theme = useTheme();
+    const [isMilitaryTimeFormat, setIsMilitaryTimeFormat] = useState(hasMilitaryTimeFormat);
+    const serverUrl = useServerUrl();
+    const intl = useIntl();
+
+    const styles = getStyleSheet(theme);
+
+    const saveButton = useMemo(() => {
+        return {
+            id: SAVE_CLOCK_BUTTON_ID,
+            enabled: false,
+            showAsAction: 'always' as const,
+            testID: 'settings_display.save.button',
+            color: theme.sidebarHeaderTextColor,
+            text: intl.formatMessage({id: 'settings.display.militaryClock.save', defaultMessage: 'Save'}),
+        };
+    }, [theme.sidebarHeaderTextColor]);
+
+    const canSave = useCallback((enabled: boolean) => {
```

Comment:
As with other PRs, I think this work better as an effect.

Reply:
corrected